crawl-ref/source/spl-transloc.cc

```diff
@@ -1469,9 +1469,17 @@ static bool _can_move_mons_to(const monster &mons, coord_def pos)
  */
 void attract_monsters()
 {
+    //get list of near monsters
+    vector<monster *> targets;
     for (monster_near_iterator mi(you.pos(), LOS_NO_TRANS); mi; ++mi)
-    {
-        if (!_can_beckon(**mi) || mi->friendly())
+        targets.push_back(*mi);
```

Comment:
```suggestion
        if (!_can_beckon(**mi) || mi->friendly())
            targets.push_back(*mi);
```

Filter valid targets now, before you sort.